### PR TITLE
Work project

### PR DESCRIPTION
**Pull request by AndriiPryslupskyi**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/satori/go.uuid"
+
+	pb "github.com/ASeegull/SmartFridge/protoStruct"
+	log "github.com/sirupsen/logrus"
+
```
Comment (anastasiia-skliar) [inline agent/agent.go:14]: This three libraries should be on the top, with other  standard libraries

File: agent/agent.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	dialer := websocket.Dialer{}
+	dialer.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	conn, resp, err := dialer.Dial(websocketURL, nil)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	defer resp.Body.Close()
+
+	fmt.Println(resp.Status)
```
Comment (anastasiia-skliar) [inline agent/agent.go:41]: Use logrus pls, and more meaningful message

File: agent/agent.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+	go timeReader(conn)
+
+	return writeInfo(conn)
+}
+
+func setToken(urlForToken string) error {
+	containerID = getID()
+	req := pb.Request{AgentID: containerID}
+	fmt.Println("container ID: ", req.AgentID)
```
Comment (anastasiia-skliar) [inline agent/agent.go:51]: Use logrus pls, and more meaningful message
Here and below

File: agent/agent.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	if err != nil {
+		return err
+	}
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return err
+	}
+
+	defer response.Body.Close()
```
Comment (anastasiia-skliar) [inline agent/agent.go:75]: No need for defer, as it is already end of function

File: agent/agent.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+	return setupInfo.UnmarshalToStruct(body)
+}
+
+func writeInfo(conn *websocket.Conn) error {
+	agentInfo := foodAgentGenerator()
+
+	for {
+		time.Sleep(time.Second * time.Duration(setupInfo.Heatrbeat))
+		agentInfo.StateExpires = int32(time.Duration(time.Now().Nanosecond()))
+		date, err := agentInfo.MarshalStruct()
```
Comment (anastasiia-skliar) [inline agent/agent.go:86]: I suppose data, not date )

File: cmd/server/main.go [hunk 6]
```diff
@@ -1,10 +1,29 @@
...
+	"github.com/ASeegull/SmartFridge/server/database"
+
+	log "github.com/sirupsen/logrus"
 )
 
 func main() {
-	fmt.Println(examplePackage.Func("server"))
+	cfg, err := config.ReadConfig()
+	if err != nil {
+		log.Fatalf("Cannot read config from file with error : %v", err)
```
Comment (anastasiia-skliar) [inline cmd/server/main.go:14]: Fatal run os.Exit after printing message. 
No need to return afterwards

File: server/handler.go [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+)
+
+//mock data
+const (
+	defaultHearBeat = 3
+	adminID         = "9079744c-ab87-4083-8400-19c14628c26f"
+)
+
+func sendErrorMsg(w http.ResponseWriter, err error, status int) {
+	log.Debug(err)
```
Comment (anastasiia-skliar) [inline server/handler.go:26]: This error will be logged twice

File: server/handler.go [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+		}
+
+		if err = database.SaveState(agentState); err != nil {
+			err = errors.New("save to db problem: ")
+			log.Println(err)
+			return
+		}
+
+		if err := conn.WriteMessage(websocket.TextMessage, []byte(fmt.Sprintf("%s %s", time.Now().Format("2-Mon-Jan-2006-15:04:05"), "Ok!"))); err != nil {
+			log.Debug(err)
```
Comment (anastasiia-skliar) [inline server/handler.go:145]: Errors should logged in with Error log level, 
Here and in other places

**Review by anastasiia-skliar — APPROVED** [review]

**Review by oleksandrkravchuk87 — APPROVED** [review]

File: server/handler.go (hunk 7 above)
Comment (AndriiPryslupskyi) [reply server/handler.go:26]: But I don't use logger before using this method

Comment (anastasiia-skliar) [reply server/handler.go:26]: Lets look at example:
```
+	foods, err := userID.GetFoodsInFridge()
 +	if err != nil {
 +		sendErrorMsg(w, err, http.StatusInternalServerError)
 +		return
 +	}
```
Both GetFoodsInFridge and sendErrorMsg will log error
It is not critical comment at all, but pay attantion

**Pull request merged**